eduNEXT/xblock-mindmap
# feat: consider weight in score calculation

### Description
This PR changes what is considered the final score for the student. Before these changes, we considered the `raw score` as the final score for the student without considering the weight configured. Now, we calculate the final score based on the weight configured in studio. This change has a downside: after calculating the new score, we could get numbers like 5.3 which cannot be saved in the submissions API so we decided to round to the nearest integer. 

### How to test
0. Configure grading in your course as follows:
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/3b6d80c3-28f1-4c26-ab5e-457b96b77821)
1. Install this branch and add the xblock to your course
2. Configure your mindmap with your preferred weight. For this test, we're using 10 (the default):
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/7dcea92a-ed8a-4442-a250-6ace6948888b)
3. As a student, submit a mindmap
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/cd0bb297-f11e-414a-8c53-ad0bd1c53456)
4. As course staff, go to the component in the LMS and grade the submission:
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/475b0743-d0b7-4d84-ba04-aa192b8a7df7)
5. As a student, check out your grade:
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/de40f219-febb-4e21-a5aa-e0cff032b0be)
From the progress page:
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64440265/9e1654c7-5071-4faa-a348-0d6538041b55)
Since the problem weight is 100, then our grade is 50. If the problem weight is 50, then our grade would be 25% and so on. Now, the subsection grade is our weighted score.
